rrule/src/iter/rruleset_iter.rs

```diff
@@ -70,32 +75,33 @@ impl<'a> RRuleSetIter<'a> {
         Ok(date)
     }
 
-    // TODO should this be called `accept` or `except`?
-    fn accept_generated_date(
+    fn is_generated_date_excluded(
         date: &Option<DateTime>,
         exrules: &[RRule],
-        exdates: &mut HashMap<i64, ()>,
+        exdates: &mut BTreeSet<i64>,
+        dt_start: &DateTime,
     ) -> bool {
         match date {
-            None => true,
+            None => false,
             Some(date) => {
-                let dt = date.timestamp();
+                let ts = date.timestamp();
 
                 if !exrules.is_empty() {
-                    let after = date.timezone().timestamp(dt - 1, 0);
-                    let before = date.timezone().timestamp(dt + 1, 0);
+                    let start = date.timezone().timestamp(ts - 1, 0);
```

Comment:
I don't know what we are doing here exactly.

Reply:
The function `is_generated_date_excluded` is called with a date-time that was generated from an `RRULE` or `RDATE` and checks if it matches against an `EXDATE` or `EXRULE`, in which case the date-time will be discarded. The before and after timestamps are used as boundaries when checking the `EXRULE:` iterator for a match against the given date-time so that we don't have to iterate through every occurrence in the `EXRULE` when we know that it has to be in the interval <ts - 1, ts + 1>.

Reply:
While you are here, do you mind cleaning up the return code for the function? :)
```rust
if exdates.contains(&ts) {
  return true;
}

false
```

To
 
```rust
exdates.contains(&ts)
```

Reply:
Strange that clippy didn't catch this! 🤔

Reply:
Mmm, I still don't get the logic there.
And I found some infinite loops there, I think.
I put it in my future PR:

```
#[test]
fn edge_case_1() {
    let rrule: RRuleSet = "DTSTART;TZID=Europe/Berlin:20210101T000000;
RRULE:FREQ=MONTHLY
EXRULE:FREQ=MONTHLY"
        .parse()
        .expect("The RRule is not valid");

    rrule.into_iter().next();
}
```